app/scripts/controllers/attachPVC.js

```diff
@@ -175,24 +238,27 @@ angular.module('openshiftConsole')
             var readOnly = $scope.attach.readOnly;
             if (mountPath) {
               // for each container in the pod spec, add the new volume mount
-              angular.forEach(podTemplate.spec.containers, function(container) {
-                if (isContainerSelected(container)) {
-                  var newVolumeMount =
-                    StorageService.createVolumeMount(name, mountPath, subPath, readOnly);
-                  if (!container.volumeMounts) {
-                    container.volumeMounts = [];
-                  }
-                  container.volumeMounts.push(newVolumeMount);
-                }
-              });
+              // angular.forEach(podTemplate.spec.containers, function(container) {
+              if(!setVolumeMount(podTemplate, name, mountPath, subPath, readOnly)) {
+                $scope.disableInputs = false;
+                return;
+              }
             }
 
             // add the new volume to the pod template
             var newVolume = StorageService.createVolume(name, persistentVolumeClaim);
             if (!podTemplate.spec.volumes) {
               podTemplate.spec.volumes = [];
             }
-            podTemplate.spec.volumes.push(newVolume);
+
+            // if the newly created volume already exists, only
+            // fail if the "overwrite" option was not set
+            var volumeExists = _.some(podTemplate.spec.volumes, { name: newVolume.name });
+
+
+            if (!$scope.attach.overwrite || ($scope.attach.overwrite && !volumeExists)) {
+              podTemplate.spec.volumes.push(newVolume);
+            }
```

Comment:
Is it correct to continue if the volume exists and overwrite is not set?

Reply:
> Is it correct to continue if the volume exists and overwrite is not set?

Ah, no pushing a volume containing a duplicate name field would cause an error, however I think I was relying on the client-side checks that we currently have to stop a user from submitting the form if a duplicate name or mount path was found.

Will go ahead and update this to just be 
```js
if (!volumeExists) {
  podTemplate.spec.volumes.push(newVolume);
}
```